### PR TITLE
Apply standard and cleanup of Measure-DbaDiskSpaceRequirement

### PR DESCRIPTION
**Pull request by wsmelton**
General cleanup of standards.

Addressed variable and properties names to make less generic and a bit clearer in reading output.

Removed use of EnableException parameter on Write-Message and Stop-Function calls.

Added code block for building connections to source and destination, to allow it to also properly test the access to each before trying to run the code.

@FriedrichWeinmann and/or @potatoqualitee  please review...if there is anything that needs to be addressed let me know (or by all means address, the branch is in the dbatools repo).

### CODE REVIEW

File: functions/Measure-DbaDiskSpaceRequirement.ps1 [hunk 1]
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
...
         [string]$SourceDatabase,
         [PSCredential]$SourceSqlCredential,
-        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
         [DbaInstanceParameter]$Destination,
-        [Parameter(Mandatory=$false,ValueFromPipelineByPropertyName)]
+        [Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true)]
         [string]$DestinationDatabase,
         [PSCredential]$DestinationSqlCredential,
         [PSCredential]$Credential
```
Comment (potatoqualitee) [inline functions/Measure-DbaDiskSpaceRequirement.ps1]: Needs EnableException added, as well as docs

File: functions/Measure-DbaDiskSpaceRequirement.ps1 [hunk 2]
```diff
@@ -24,20 +23,27 @@ Function Measure-DbaDiskSpaceRequirement {
...
+            To connect as a different Windows user, run PowerShell as that user.
 
         .PARAMETER DestinationDatabase
-            The database name at destination. May or may not be present. Unspecified name will assume database name be the same as source.
+            The database name at destination.
+            May or may not be present, if unspecified it will default to the database name provided in SourceDatabase.
 
         .PARAMETER Credential
-            Windows credentials to connect via CIM/WMI/PowerShell remoting for MountPoint definition.
+            The credentials to use to connect via CIM/WMI/PowerShell remoting.
```
Comment (potatoqualitee) [inline functions/Measure-DbaDiskSpaceRequirement.ps1]: Add Docs for EnableException

File: functions/Measure-DbaDiskSpaceRequirement.ps1 [hunk 3]
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
...
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
         [DbaInstanceParameter]$Destination,
-        [Parameter(Mandatory=$false,ValueFromPipelineByPropertyName)]
+        [Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true)]
         [string]$DestinationDatabase,
         [PSCredential]$DestinationSqlCredential,
         [PSCredential]$Credential
     )
     begin {
-        $NullText = '#NULL'
```
Comment (potatoqualitee) [inline functions/Measure-DbaDiskSpaceRequirement.ps1]: this should just be $null in the output - I'm not aware of any value in #NULL

File: functions/Measure-DbaDiskSpaceRequirement.ps1 [hunk 4]
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
...
     #>
     [CmdletBinding()]
-    Param(
-        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+    param(
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
         [DbaInstanceParameter]$Source,
-        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
         [string]$SourceDatabase,
```
Comment (potatoqualitee) [inline functions/Measure-DbaDiskSpaceRequirement.ps1]: Please change to just Database - I did a mass replace for SourceDatabase -> Database and it worked well, including docs

File: functions/Measure-DbaDiskSpaceRequirement.ps1 [hunk 5]
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
...
-                    FilePath2 = $NullText
-                    SizeKB1 = $File1.Size
-                    SizeKB2 = 0
-                    DiffKB = $File1.Size
-                    ComputerName = $ComputerName
-                    MountPoint = Get-MountPointFromDefaultPath -DefaultPathType $File1.Type -SqlInstance $Destination -SqlCredential $DestinationSqlCredential
-                })
+            if (!$found) {
+                # Files on source but not on destination
+                $details += @([PSCustomObject]@{
```
Comment (potatoqualitee) [inline functions/Measure-DbaDiskSpaceRequirement.ps1]: this should be released to the pipeline immediately and not collected

File: functions/Measure-DbaDiskSpaceRequirement.ps1 [hunk 6]
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
...
+                    })
             }
         }
         $DestinationDatabase = ''
     }
     end {
-        $Detail | Select-DefaultView -Property DatabaseName1, DatabaseName2, Name1, Name2, SizeKB1, SizeKB2, DiffKB, ComputerName, MountPoint
+        if (Test-FunctionInterrupt) { return }
+
+        $details | Select-DefaultView -Property DatabaseName1, DatabaseName2, Name1, Name2, SizeKB1, SizeKB2, DiffKB, ComputerName, MountPoint
```
Comment (potatoqualitee) [inline functions/Measure-DbaDiskSpaceRequirement.ps1]: Unfortunately, I didn't see much in the output for this PR. Something went wrong somewhere.

File: functions/Measure-DbaDiskSpaceRequirement.ps1 [hunk 7]
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
...
+                        DiffKB        = $destFileNotSource.Size * -1
+                        MountPoint    = Get-MountPointFromPath -Path $destFileNotSource.Filename -ComputerName $computerName -Credential $Credential
+                    })
             }
         }
         $DestinationDatabase = ''
     }
     end {
-        $Detail | Select-DefaultView -Property DatabaseName1, DatabaseName2, Name1, Name2, SizeKB1, SizeKB2, DiffKB, ComputerName, MountPoint
+        if (Test-FunctionInterrupt) { return }
```
Comment (potatoqualitee) [inline functions/Measure-DbaDiskSpaceRequirement.ps1]: Not required at this stage

File: functions/Measure-DbaDiskSpaceRequirement.ps1 [hunk 8]
```diff
@@ -70,186 +76,224 @@ Function Measure-DbaDiskSpaceRequirement {
...
+                            DestinationComputerName = $destServer.NetName
+                            DestinationInstance     = $destServer.ServiceName
+                            DestinationSqlInstance  = $destServer.DomainInstanceName
+                            SourceDatabase          = $sourceDb.Name
+                            DestinationDatabase     = $destDb.Name
+                            SourceLogicalName       = $sourceFile.Name
+                            DestinationLogicalName  = $destFile.Name
+                            SourceFileName          = $sourceFile.FileName
+                            DestinationFileName     = $destFile.FileName
+                            SourceFileSizeKB        = $sourceFile.Size
```
Comment (potatoqualitee) [inline functions/Measure-DbaDiskSpaceRequirement.ps1]: Can we use [dbasize] here?

Comment (potatoqualitee) [reply functions/Measure-DbaDiskSpaceRequirement.ps1]: And anywhere else relevant. may require changing to bytes

Comment (wsmelton) [reply functions/Measure-DbaDiskSpaceRequirement.ps1]: This is just returning the raw value, which is in KB. I can see how the DB space command handles it and just match that.

**Comment by wsmelton** [comment]
@potatoqualitee if you get a chance give this one a spin, fixed the output issue as I had to update properties on the call to default view...forgot to update those.

I added DbaSize and think I did this right, the size unit from SMO is in KB so I simply had it product of that value times 1k and that should be the value in Bytes.

**Comment by potatoqualitee** [comment]
sweet will do, thank you shawn. sorry for the delay, the past 3 days have been crazy!

**Comment by potatoqualitee** [comment]
LOVE! such a cool command. thanks for the updates @wsmelton and thanks for the original command @pollusb !!

image

**Pull request merged**